docs/spec/spec.md

```diff
@@ -0,0 +1,202 @@
+# Object Model
+
+Knative ships with several objects which implement useful sets of
+[interfaces](interfaces.md). It is expected that additional objects which
+implement other sets of interfaces will be added over time. For context, see
+the [overview](overview.md) and [motivations](motivation.md) sections.
+
+These are Kubernetes resources that been introduced using Custom Resource
+Definitions. They will have the expected _ObjectMeta_, _Spec_, _Status_ fields.
+This document details our _Spec_ and _Status_ customizations.
+
+- [Channel](#kind-channel)
+- [Subscription](#kind-subscription)
+- [Provider](#kind-provisioner)
+
+## kind: Channel
+
+### group: eventing.knative.dev/v1alpha1
+
+_A Channel logically receives events on its input domain and forwards them to
+its subscribers. Additional behavior may be introduced by using the
+Subscription's call parameter._
+
+### Object Schema
+
+#### Spec
+
+| Field         | Type                               | Description                                                                | Limitations |
+| ------------- | ---------------------------------- | -------------------------------------------------------------------------- | ----------- |
+| provisioner\* | ProvisionerReference               | The name of the provisioner to create the resources that back the Channel. | Immutable.  |
+| arguments     | runtime.RawExtension (JSON object) | Arguments to be passed to the provisioner.                                 |             |
+
+\*: Required
+
+#### Metadata
+
+##### Owner References
+
+- Owned (non-controlling) by the ClusterProvisioner used to provision the
+  Channel.
+
+#### Status
+
+| Field      | Type       | Description                                                                                                                 | Limitations |
+| ---------- | ---------- | --------------------------------------------------------------------------------------------------------------------------- | ----------- |
+| sinkable   | Sinkable   | Address to the endpoint as top-level domain that will distribute traffic over the provided targets from inside the cluster. |             |
```

Comment:
There is no type `Sinkable`, I presume the intent is to have a `domainInternal` field that is a dns name

Reply:
ah, I missed that these had already been implemented in ducks/v1alpha1 and consumed by serving/v1alpha1.

This implies that a KService can never act as an event sink. Is that our intention?

I'm unsure if it is worth distinguishing between Sinkable and Targetable in the duck types. The difference between the two is much more about how the caller behaves than how the callee behaves.

Reply:
that's a good point. My only thinking is that even though they behave the same at the control plane, because they have different semantics at the data plane it might be in our best interest in the long term to separate the two into different objects. @evankanderson has been thinking about inheritance possibly here. I frankly am not sure which way is better, it's just my gut that just because they have the same behaviour at this level putting the two into a single interface might be restricting in the future. Seems safer to keep them separate and if the experience shows there's no difference in real world to combine them than have to split them apart later. Thoughts?